lib/elixir/lib/protocol.ex

```diff
@@ -569,8 +569,8 @@ defmodule Protocol do
 
         unquote(block)
 
-        Module.register_attribute(__MODULE__, :impl, persist: true)
-        @impl [protocol: @protocol, for: @for]
+        Module.register_attribute(__MODULE__, :protocol_metadata, persist: true)
+        @protocol_metadata [protocol: @protocol, for: @for]
```

Comment:
@josevalim Do you think we should document this somewhere? It seems unlikely but plausible that a user could accidentally use this module for their own purposes and be very surprised at the resulting breakage.

For completeness I think it would be nice to list all the "blacklisted" module variables somewhere.